docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md

```diff
@@ -0,0 +1,254 @@
+
+This project needs to complete the task definition part and the unknown task identification part.
+
+Task definition is the predecessor task, the business model is trained by the RFNet semantic segmentation algorithm, and the datasets involved are Cityscapes (camera data) and SYNTHIA-RAND-CITYSCAPES (simulation data).
+
+Unknown task identification is the core task of this project. The unknown task recognition is data inference, using Wide-ResNet network structure, combined with the classification placeholders and data placeholders in CVPR2021 paper "Learning placeholders for open-set recognition" to form a neural network and train the classifier, which can classify known samples and identify the unknown samples.
+
+### 3.1 Architecture
+
+This project mainly implements the function of discovering new tasks.
+
+<img src="images/Lifelong_learning_structure.png" alt="Lifelong_learning_structure" style="zoom:48%;" />
+
+The following is the architecture diagram of this project system, as shown in the figure, the input of the unknown sample identification module is the inference dataset, and the output bits are known sample set and unknown sample set. Before the unknown sample identification module is the task definition module, after the task definition, the initialized dataset is considered as known dataset. In the task assignment module after the unknown sample identification module, the known data are inferred and the unknown data are re-posted and incorporated into the edge-side knowledge base and then into the cloud-side knowledge base.
+
+<img src="images/Algorithm flow.png" style="zoom:48%;" />
+
+### 3.2 Usecase
+
+#### 3.2.1 Calling datasets
+
+```python
+from sedna.datasources import BaseDataSource
+```
+
+#### 3.2.2 User flow chart
+
+<img src="images/User.png" alt="User" style="zoom:48%;" />
+
+Usage Process
+1. View the dataset
+2. download the edge cloud synergy dataset
+3. Package your algorithm into Estimator
+4. Call the Benchmark module, and the Benchmark module calls the developer's encapsulated Estimator
+5. Run and view the results
+6. Update and view the local edge cloud collaborative AI ranking
+
+#### 3.2.3 Interface Description
+
+```yaml
+type: "unknown_task_recognition"
+name: "UnknownTaskRecognition"
+url: "./examples/curb-detection/lifelong_learning_bench/testalgorithms/rfnet/unknown_task_recognition.py"
+hyperparameters:
+      # name of the hyperparameter; string type;
+      # dataset which should be the dataset you chose to call above.
+   - dataset:
+     values:
+       - "dataset_test"
+```
+
+
+
+## 4 datasets
+
+In this project, the image source is used as the classification criterion, and RGB images in three datasets are selected as samples, among which, the real camera dataset and simulator dataset are selected for the known task dataset, and the animation dataset of Akiyama open source is selected for the unknown task dataset. The sample dataset and dataset structure are shown below.
```

Comment:
To fit a real world setting, it is recommended to set the real camera dataset as unknown.

Reply:
It seems that this comment is resolved.

Reply:
Yep, this comment has been modified in the experiment, but this version of the proposal has not been modified.